instagram/src/components/PostContainer/PostPage.js

```diff
@@ -0,0 +1,105 @@
+import React, { Component } from 'react';
+import Loader from 'react-loader-spinner';
+import dummyData from '../../dummy-data';
+import PostContainer from '../PostContainer/PostContainer';
+import SearchBar from '../SearchBar/SearchBar';
+
+export default class PostPage extends Component {
+    constructor() {
+        super();
+        this.state = {
+            liked: false,
+            searchValue: '',
+            data: []
+        };
+    }
+
+    componentDidMount() {
+        this.setState({
+            data: localStorage.getItem('data') === null ? dummyData : JSON.parse(localStorage.getItem('data'))
+        });
+    }
+
+    submitComment = (timestamp, comment) => {
+        const commentMatch = this.state.data[this.state.data.map((obj, i) => obj.timestamp === timestamp && i).filter(arr => arr !== false).pop()];
+
+        commentMatch.comments.push(comment);
+
+        localStorage.setItem('data', JSON.stringify(this.state.data))
+    }
+
+    search = e => {
+        this.setState({ [e.target.name]: e.target.value });
+
+        const search = this.state.data.filter(post => post.username.toLowerCase().includes((e.target.value).toLowerCase()) || post.comments.map(comm => comm.text.toLowerCase().includes((e.target.value).toLowerCase()) || comm.username.toLowerCase().includes((e.target.value).toLowerCase())) === true);
+
+        search.length !== 0 && this.setState({ data: search });
+
+        if (e.target.value.length === 0) {
+            this.setState({ data: dummyData });
+        }
+    }
+
+    deleteComment = (postIndex, text) => {
+
+        let clone = this.state.data.slice();
+
```

Comment:
The other way to do this is:
```
let clone = [...this.state.data];
```
I see more codes written using spread operator than `.slice` nowadays, but it's preference... you can keep using slice too